test/metabase/api/cloud_migration_test.clj

```diff
@@ -0,0 +1,87 @@
+(ns ^:synchronized metabase.api.cloud-migration-test
```

Comment:
what's the `:synchronized` metadata is for?

Reply:
To not run the test as parallel. These tests can set the `read-only-mode` setting, which will block DB writes for a lot of different tables. This would interfere with other tests.

Reply:
oh in that case I don't think you need the flag because all tests are synchronized by default

Reply:
Uhm, are you absolutely sure about that? I see other `^:synchronized` tests, and I also see uses of `mt/with-dynamic-redefs` that I'm pretty sure are so that `with-redefs` work with parallel tests (we had this on cloud too at one point).

Reply:
well, if you look into the our test runner, it only checks the parallel metadata and it's how it knows which tests should be paralleled: https://github.com/metabase/hawk/blob/490c9b04fbf5861dbd47c0c5281dad2d113bd285/src/mb/hawk/parallel.clj

i don't know where is that synchronized is used at all.

Reply:
You know, that's a good point. I think I was remembering [eftest/synchronized](https://github.com/weavejester/eftest/blob/5a367d5ebd5fef2062b310d7c4602fdc4785edde/eftest/src/eftest/runner.clj#L22-L24), which hawk is based on. The one I used back at harbormaster was actually `^{:parallel false}`, I even made a [PR](https://github.com/metabase/hawk/commit/a8f3e8a6a53e39357863e33c25506ad7b0028558) to support it at the individual test level.

But neither of those is just `synchronized`. Yet the MB codebase has it in a few tests e.g. [this mongo one](https://github.com/metabase/metabase/blob/585b4ea46b37bdc0d3d5aaafd3377137d21efdee/modules/drivers/mongo/test/metabase/driver/mongo/query_processor_test.clj#L191), and even on a [clj-kondo custom rule](https://github.com/metabase/metabase/blob/585b4ea46b37bdc0d3d5aaafd3377137d21efdee/.clj-kondo/hooks/clojure/test.clj#L135-L137). 

At the same time I see `mt/with-dynamic-redefs` used on tests that don't seem to opt-in to parallelism at all, e.g. [this one](https://github.com/metabase/metabase/blob/585b4ea46b37bdc0d3d5aaafd3377137d21efdee/enterprise/backend/test/metabase_enterprise/upload_management/api_test.clj#L82-L89). I don't see the test runner in deps.edn set parallelism by default either.

So I agree this is a bit of a mystery. I don't want to take on solving this mystery while I'm trying to get this feature in though, so I think the right thing for me to do is remove the metadata like you suggested.